File: go/0246_ strobogrammatic_number.go
func isStrobogrammatic(num string) bool {
	nlen := len(num)
	if nlen == 0 {
		return true
	} else if nlen == 1 {
		return checkOne(num[0])
	}

	left, right := 0, nlen-1
	for left < right {
		if !check(num[left], num[right]) {
			return false
		}
		left++
		right--
	}

	if left == right {
		return checkOne(num[left])
	}
	return true
}

func check(b1, b2 byte) bool {
	switch b1 {
	case '0':
		return b2 == '0'
	case '1':
		return b2 == '1'
	case '6':
		return b2 == '9'
	case '8':
		return b2 == '8'
	case '9':
		return b2 == '6'
	}
	return false
}

func checkOne(b1 byte) bool {
	return b1 == '0' || b1 == '1' || b1 == '8'
}

File: python/0246_strobogrammatic_number.py
class Solution:
    def isStrobogrammatic(self, num: str) -> bool:
        nlen = len(num)
        if nlen == 0:
            return False

        left = 0
        right = nlen - 1
        while left <= right:
            if left == right:
                return num[left] == '0' or num[left] == '1' or num[left] == '8'

            if not self.isOK(num[left], num[right]):
                return False
            left += 1
            right -= 1
        return True

    def isOK(self, ch1, ch2):
        if ch1 == '0' and ch2 == '0':
            return True
        elif ch1 == '1' and ch2 == '1':
            return True
        elif ch1 == '6' and ch2 == '9':
            return True
        elif ch1 == '8' and ch2 == '8':
            return True
        elif ch1 == '9' and ch2 == '6':
            return True
        return False
